packages/gatsby/content/advanced/pnp-api.md

```diff
@@ -0,0 +1,122 @@
+The `topLevel` object is a simple package locator pointing to the top-level package of the dependency tree. Note that even when using workspaces you'll still only have one single top-level for the entire project.
+
+This object is provided for convenience and doesn't necessarily needs to be used; you may create your own top-level locator by setting both fields to `null`.
+
+### `getPackageInformation(...)`
+
+```ts
+export function getPackageInformation(locator: PackageLocator): PackageInformation;
+```
+
+The `getPackageInformation` function returns all the information stored inside the PnP API for a given package. The information currently available are:
+
+- `packageLocation`, which will contain the location of the package on the disk
+- `packageDependencies`, which will contain the set of dependencies for the given package
+
+### `findPackageLocator(...)`
+
+```ts
+export function findPackageLocator(location: string): PackageLocator | null;
+```
+
+Given a location on the disk, the `findPackageLocator` function will return the package that "owns" the path. For example, running this function on something conceptually similar to `/.../node_modules/foo/index.js` would return a package locator pointing to the `foo` package (and its exact version).
+
+Note that it's important that you don't call `realpath` on the path argument. The current Yarn implementation uses symlinks in order to disambiguate packages that list peer dependencies, and calling `realpath` would cause the PnP API to lose track of which set of dependencies a given path is meant to access.
+
+### `resolveToUnqualified(...)`
+
+```ts
+export function resolveToUnqualified(request: string, issuer: string | null, opts?: {considerBuiltins?: boolean}): string | null;
+```
+
+The `resolveToUnqualified` function is maybe the most important function exposed by the PnP API. Given a request (which may be a bare specifier like `lodash`, or an relative/absolute path like `./foo.js`), the PnP API will return the unqualified resolution.
+
+For example, the following:
+
+```
+lodash/uniq
+```
+
+Might very well be resolved into:
+
+```
+/my/cache/lodash/1.0.0/node_modules/lodash/uniq
+```
+
+### `resolveUnqualified(...)`
+
+```ts
+export function resolveUnqualified(unqualified: string, opts?: {extensions?: Array<string>}): string;
+```
+
+The `resolveUnqualified` function is mostly provided as an helper; it reimplements the Node resolution for file extensions and folder indexes, but not the regular `node_modules` traversal. It makes it slightly easier to integrate PnP into some projects, although it isn't required in any way if you already have something that fits the bill.
+
+To give you an example `resolveUnqualified` isn't needed with `enhanced-resolved`, used by Webpack, because it already implements its own way the logic contained in `resolveUnqualified` (and more). Instead, we instead only have to leverage the lower-level `resolveToUnqualified` function and feed it to the regular resolver.
```

Comment:
```suggestion
To give you an example `resolveUnqualified` isn't needed with `enhanced-resolved`, used by Webpack, because it already implements its own way the logic contained in `resolveUnqualified` (and more). Instead, we only have to leverage the lower-level `resolveToUnqualified` function and feed it to the regular resolver.
```

Reply:
😅